opennms-services/src/main/java/org/opennms/netmgt/threshd/ThresholdingSet.java

```diff
@@ -337,13 +341,18 @@ protected boolean passedThresholdFilters(CollectionResourceWrapper resource, Thr
                     final Matcher m = p.matcher(attr);
                     boolean pass = m.matches();
                     LOG.debug("passedThresholdFilters: the value of {} is {}. Pass filter? {}", f.getField(), attr, pass);
+
                     if (operator.equals("or") && pass) {
-                        return true;
+                        LOG.debug("passedThresholdFilters: Filter #{} passed and logical connection operator is 'or', no need to apply other filters", count);
+                        break;
```

Comment:
Why are you adding a "break" statement here ? If the operator was "or", obviously it will never be equal to "and" ;)